### ISSUE TITLE
Opening file with special content on WSL2 filesystem from Windows 10 takes very long time

### ISSUE THREAD

**Issue by kosi2801** [microsoft/WSL#9504]
### Version

Microsoft Windows [Version 10.0.19044.2486]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.15.79.1-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Wed Nov 23 01:01:46 UTC 2022

### Distro Version

Kali Linux 2022.4; Ubuntu 22.04

### Repro Steps

Create a file with special binary contents inside WSL2. The attached [xae.zip](https://github.com/microsoft/WSL/files/10449633/xae.zip) contains a truncated portion of the file https://repo1.maven.org/maven2/xpp3/xpp3_min/1.1.4c/xpp3_min-1.1.4c.jar in binary form.

1. Download the ZIP inside WSL2 distro, unzip to get the file "xae"
2. In Windows open an Explorer window and go to the location of the file, eg "\\wsl$\kali-linux\home\user\"
3. Right-click the file "xae" and "Open With" with eg. Notepad

### Expected Behavior

Notepad opens instantly showing binary contents of file "xae"

### Actual Behavior

First time trying to open the file: Notepad starts, but blocks for a very long time (~120sec) before displaying the file content.
Subsequent attempts: in 90% of the attempts Notepad starts and displays the file immediately, sometimes it still blocks again.

This happens in both Linux variants that I tried (Kali, Ubuntu) and the behavior started appearing in the second week of 2023. Maybe after a Windows update, but this cannot be traced back exactly anymore.

I could not reproduce the issue by manually creating textfiles (up to 8MB) on Linux and opening them, it seems to only happen with binary files. It's very reproducable in a local Java Maven repository (~/.m2) where almost every .jar that is attempted to be opened from Windows causes the opening process to block.
It doesn't matter what application tries to access the file, it also happens when copying the file from \\wsl$\.... to a Windows directory.
Also, the larger the .jar file the longer the blocking behavior it seems.

**Comment by kosi2801** [comment]
Created [WslLogs-2023-01-19_08-13-57.zip](https://github.com/microsoft/WSL/files/10454414/WslLogs-2023-01-19_08-13-57.zip).
Logs contain copying the file "xae" inside WSL2 (Kali), modifying two bytes with editor (to workaround cachings) and opening the resulting file "xae_1" with Windows notepad from \\wsl.localhost\kali-linux\home\ahoefler\.m2_local\tst\xae_1.

**Comment by kosi2801** [comment]
I now also tried with the WSL 1.1.0 pre-release version from https://github.com/microsoft/WSL/releases/tag/1.1.0, using the 5.15.83.1-microsoft-standard-WSL2 kernel version. Still same behavior.

**Comment by Kas-tle** [comment]
I am not entirely sure if my issue is the same... I have found that when I try to copy jar files specifically from WSL to Windows the copy process hangs... in file explorer it will never finish. Interestingly, using terracopy hangs for about 45 seconds but eventually does succeed at copying the file to Windows. I have also noticed that compressing the file into a zip seems to bypass this... so I wonder if this is some sort of Windows Defender shenanigans.
